src/WarcraftLegacies.Source/Setup/FactionSetup/DalaranSetup.cs

```diff
@@ -91,6 +91,8 @@ public static void Setup(PreplacedUnitSystem preplacedUnitSystem)
       Dalaran.ModObjectLimit(FourCC("R06O"), Faction.UNLIMITED); //Improved Phase Blade
       Dalaran.ModObjectLimit(FourCC("R00J"), Faction.UNLIMITED); //Combat Tomes
 
+      Dalaran.ModObjectLimit(Constants.UPGRADE_R0A7_ESCAPE_TO_THERAMORE_DALARAN, Faction.UNLIMITED);
```

Comment:
This should be in the `OnAdd` method of `QuestTheramore`, so that it's attached to the quest rather than the faction. Check out the `QuestOgrimmar` quest for a good example.